client/disputes/index.tsx

```diff
@@ -314,7 +320,10 @@ export const DisputesList = (): JSX.Element => {
 							);
 							const history = getHistory();
 							history.push(
-								getDetailsURL( dispute.dispute_id, 'disputes' )
+								getDetailsURL(
+									dispute.charge_id,
+									'transactions'
+								)
```

Comment:
I guess the follow up question is… is it a bad idea to switch all these to `getAdminUrl` now as part of fixing this?

Reply:
Interestingly, I made the choice to change one of these links from `getAdminUrl` → `getDetailsURL` since I assumed it was the preferred option for detail screen URLs.

I'm not sure if that's a bad idea.

Reply:
It all depends on what `parentSegment` means, and what value it adds.

Reply:
> I assumed it was the preferred option for detail screen URLs.

What do you prefer and why?

Reply:
To complicate things, `getAdminUrl` is a wrapper for `addQueryArgs` – should this be used instead of the wrapper function?

https://github.com/Automattic/woocommerce-payments/blob/2e47bb83b8d0a85deecfb658622d29be1b8b44eb/client/utils/index.js#L25

I think we leave this PR very specific to the scope of the PR and change links only.

Reply:
Yeah I agree, probably making extra work to switch over from `DetailsLink` / `getDetailsURL`. 

> To complicate things, getAdminUrl is a wrapper for addQueryArgs – should this be used instead of the wrapper function?

Here's my take…

- `getAdminUrl` is a useful, easy to understand API for constructing a url anywhere in Woo admin. We should use it whereever it makes sense.
- If we have specific boilerplate we end up doing a lot, it might make sense to have our own API, e.g. `getWooPaymentsAdminUrl`. 
  - `getDetailsURL` is not an improvement over `getAdminUrl` - makes the code less readable.

Reply:
Bonus points for `getAdminUrl` – if you understand how it works (what the params mean), that knowledge transfers to `addQueryArgs` (and vice versa). Not so for `DetailsLink` and `getDetailsURL`.

Reply:
> What do you prefer and why?

For this PR, I'd prefer if the changes were specific to the goal. 😆

From a code-quality perspective, I'd like to understand if having the `getDetailsUrl` with typed parameters helps us in some way before removing it.

Thinking about a developer using `getDetailsUrl`, it might be a helpful abstraction. They won't have to remember if it is `transaction/details`, `/transactions/details`, `/payments/transactions/details`, `%2Ftransactions%2Fdetails` etc. The function remembers all of this and TS yells at me if I use the wrong parent (`transaction` instead of `transactions`).

Reply:
Fair point - a typed wrapper of `getAdminUrl` or `getWooPaymentsAdminUrl` could add value, though this is orthogonal to my concern that `getDetailURL( … parentSegment )` is an unintuitve, weird API.

Reply:
I'd find it much easier to read an API like 

- `getWooPaymentsUrl( collection: 'transactions' )`
- `getWooPaymentsUrl( collection: 'transactions', item: transactionId  )`

Or more transparently

- `getWooPaymentsUrl( `transactions` )`
- `getWooPaymentsUrl( `transactions/details/${ transactionId }` )`

If you were designing this API, what would it look like?